src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -21,10 +25,17 @@ internal class ManagedIdentityClient
         private const string WindowsHimdsFilePath = "%Programfiles%\\AzureConnectedMachineAgent\\himds.exe";
         private const string LinuxHimdsFilePath = "/opt/azcmagent/bin/himds";
         internal static ManagedIdentitySource s_sourceName = ManagedIdentitySource.None;
+        internal static ITimeService s_timeService = new TimeService();
+        internal static readonly TimeSpan MtlsCertRefreshSkew = TimeSpan.FromMinutes(5);
+
+        internal static readonly ConcurrentDictionary<string,
```

Comment:
have you considered moving the cache elsewhere? It seems that this class doesn't even use it?